app/shop.py

```diff
@@ -0,0 +1,41 @@
+
+from app.extra import Point
+
+
+class Shop:
+
+    @classmethod
+    def from_dict(cls, shop_data: dict) -> list:
+        return [Shop(
+            shop["name"],
+            Point(shop["location"][0], shop["location"][1]),
+            shop["products"],
+        ) for shop in shop_data]

```

Comment:
Why classmethod? If you want to use classmethod you must use `cls` instead of `Shop`.
Or just use staticmethod